hadoop-ozone/tools/src/main/java/org/apache/hadoop/ozone/admin/failover/FailoverCommand.java

```diff
@@ -0,0 +1,414 @@
+import org.apache.hadoop.ozone.ha.ConfUtils;
+import org.apache.hadoop.ozone.om.OMConfigKeys;
+import org.apache.hadoop.ozone.om.ha.OMHANodeDetails;
+import org.apache.hadoop.ozone.om.helpers.ServiceInfoEx;
+import org.apache.ratis.client.RaftClient;
+import org.apache.ratis.client.RaftClientConfigKeys;
+import org.apache.ratis.client.api.GroupManagementApi;
+import org.apache.ratis.conf.Parameters;
+import org.apache.ratis.conf.RaftProperties;
+import org.apache.ratis.grpc.GrpcConfigKeys;
+import org.apache.ratis.grpc.GrpcTlsConfig;
+import org.apache.ratis.proto.RaftProtos;
+import org.apache.ratis.protocol.*;
+import org.apache.ratis.retry.ExponentialBackoffRetry;
+import org.apache.ratis.util.TimeDuration;
+import org.kohsuke.MetaInfServices;
+import picocli.CommandLine;
+
+import java.io.IOException;
+import java.security.cert.X509Certificate;
+import java.util.*;
+import java.util.concurrent.TimeUnit;
+import java.util.stream.Collectors;
+
+import static org.apache.hadoop.hdds.scm.ScmConfigKeys.*;
+
+/**
+ * Subcommand for admin operations related to failover actions.
+ */
+@CommandLine.Command(
+    name = "failover",
+    description = "manually transfer leadership of raft group to target node",
+    mixinStandardHelpOptions = true,
+    versionProvider = HddsVersionProvider.class
+)
+@MetaInfServices(SubcommandWithParent.class)
+public class FailoverCommand extends GenericCli
+    implements SubcommandWithParent {
+
+  public static final RaftGroupId PSEUDO_RAFT_GROUP_ID = RaftGroupId.randomId();
+  public static final int TRANSFER_LEADER_WAIT_MS = 120_000;
+  public static final String RANDOM = "RANDOM";
+
+  enum Domain {
+    OM,
+    SCM,
+  }
+
+  @CommandLine.ParentCommand
+  private OzoneAdmin parent;
+
+  @CommandLine.Parameters(
+      description = "['om'/'scm']"
+  )
+  private String domain;
+
+  @CommandLine.Parameters(
+      description = "[host/host:port/'random'] ratis rpc address of " +
+      "target domain if 'random' then choose arbitrary follower node "
+  )
+  private String tgtAddress;
+
+  @CommandLine.Option(
+      names = {"-id", "--service-id"},
+      description = "Ozone Manager Service ID, if domain is om, " +
+          "this option is prerequisite"
+  )
+  private String omServiceId;
+
+  @CommandLine.Option(
+          names = {"--hostPortList"},
+          description = "if need to test on given nodes, " +
+                  "pattern like this 'host:port,host:port,host:port'"
+  )
+  private String ratisAddresses;
+
+  public OzoneAdmin getParent() {
+    return parent;
+  }
+
+  @Override
+  public Void call() throws Exception {
+    OzoneConfiguration configuration;
+    if (parent.getOzoneConf() == null) {
+      configuration = new OzoneConfiguration();
+    } else{
+      configuration = parent.getOzoneConf();
+    }
+    try {
+      transferLeadership(configuration);
+    } catch (Exception ex) {
+      ex.printStackTrace();
+      throw ex;
+    }
```

Comment:
Ozone CLI handles uncaught exception and prints message or stack trace depending on this global option:

```
--verbose   More verbose output. Show the stack trace of the errors.
```

So the `printStackTrace()` is unnecessary here.  (And then `try-catch` can be omitted, too.)

```suggestion
    transferLeadership(configuration);
```